support/bazel/README.md

```diff
@@ -1,85 +1,63 @@
-there is an easy approach to use the {fmt} repository with Bazel out of the box.
-This is demonstrated in the following example.
+The following minimal example shows how to use {fmt} as a dependency within a Bazel project.
 
-Add to your `WORKSPACE` file:
+The following file structure is assumed:
 
-```python
-load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
-
-# Fetch all files from fmt including the BUILD file `support/bazel/BUILD.bazel`
-new_git_repository(
-    name = "fmt_workaround",
-    branch = "master",
-    remote = "https://github.com/fmtlib/fmt/",
-    build_file_content = "# Empty build file on purpose"
-)
-
-# Now the BUILD file `support/bazel/BUILD.bazel` can be used:
-new_git_repository(
-    name = "fmt",
-    branch = "master",
-    remote = "https://github.com/fmtlib/fmt/",
-    build_file = "@fmt_workaround//:support/bazel/BUILD.bazel"
-)
 ```
-
-Create a `BUILD.bazel` file and add a dependency to {fmt}:
-
-```python
-cc_binary( # Build a binary
-    name = "Demo", # Name of the binary
-    srcs = ["main.cpp"], # List of files - we only have main.cpp
-    deps = ["@fmt//:fmt"], # Depend on fmt
-)
+example
+├── BUILD.bazel
+├── main.cpp
+└── WORKSPACE.bazel
 ```
 
-Make use of {fmt} in `main.cpp`:
+*main.cpp*:
 
-```C++
+```c++
 #include "fmt/core.h"
 
 int main() {
-  fmt::print("The answer is {}.\n", 42);
+  fmt::print("The answer is {}\n", 42);
 }
 ```
 
 The expected output of this example is `The answer is 42`.
 
-## Bazelize fmt
-
-First downloading a build file and then making use of it can be considered as a bit unclean, nevertheless, it works.
-
-A cleaner Bazel solution would be to move the `WORKSPACE` and `BUILD` files to the root folder of the {fmt} Git repository.
-
-In favor of keeping the {fmt} project directory clean, those files were not added to the project root directory. 
-
-If you do not like this, you can fork this repository and move the files `BUILD.bazel`, `WORKSPACE.bazel`, `.bazelrc`, and `.bazelversion` from this folder (`support/bazel`) to the root folder of this project.
-
-This way {fmt} gets bazelized and can be used in your Bazel builds.
-
-**Example**
-
-Create a `WORKSPACE.bazel` file with the following content:
+*WORKSPACE.bazel*:
 
 ```python
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 
-# Fetch bazelized fmt
 git_repository(
     name = "fmt",
-    branch = "bazel-support", # A copy of master where BUILD.bazel, WORKSPACE.bazel, .bazelrc and .bazelversion are moved to root
-    remote = "https://github.com/<user_or_organisation>/fmt", # replace <user_or_organisation> by a valid account
+    branch = "master",
+    patch_cmds = [
+        "mv support/bazel/.bazelrc .bazelrc",
+        "mv support/bazel/.bazelversion .bazelversion",
+        "mv support/bazel/BUILD.bazel BUILD.bazel",
+        "mv support/bazel/WORKSPACE.bazel WORKSPACE.bazel",
+    ],
```

Comment:
I don't want to say that it's a bad thing, but looks like [this solution wouldn't work on Windows](https://docs.bazel.build/versions/main/repo/http.html#:~:text=the%20%60patches%60%20attribute.-,patch_cmds,be%20executed%20on%20Windows%2C%20which%20requires%20Bash%20binary%20to%20exist.,-patch_tool) without Bash.

Reply:
Good point!

The assumption here is that MSYS2 x86_64 is installed on Windows. If you follow the [Bazel Windows installation guide](https://docs.bazel.build/versions/4.2.1/install-windows.html) there is this section:

> Depending on which languages you want to build, you will need:
> 
> 
> MSYS2 x86_64
> 
> MSYS2 is a software distro and building platform for Windows. It contains Bash and common Unix tools (like grep, tar, git).
> 
> You will need MSYS2 to build, test, or run targets that depend on Bash. Typically these are genrule, sh_binary, sh_test, but there may be more (e.g. Starlark rules). Bazel shows an error if a build target needs Bash but Bazel could not locate it.
> 

Since I had on every system also MSYS2 x86_64 installed (CI job installed it) it worked without any problems.

I could now add a comment e.g. "On Windows systems make sure that MSYS2 x86_64 is installed according to the Bazel Installation guide..." and/or provide Powershell commands to move the files.

I am even not sure if the rule `git_repository` works without MSYS2 x86_64 being installed. I could imagine that without MSYS2 x86_64 there will be no `git` for Bazel in place (even if you have a "normal" Git installation on your Windows 10 system/environment path) - but this is pure speculation from my side - I have to test this first. Once I have found out how this works I will provide an update to this PR.

Reply:
Okay tested it in the meanwhile. The rule `git_repository` works without MSYS2being installed. `patch_cmds` does not work on Windows if MSYS2 is not installed. Therefore, `patch_cmds_win` has to be provided to make it working without MSYS2 installed. I fixed the PR accordingly.